audio/aacplusenc/Portfile

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+
+name                aacplusenc
+version             0.17.5
+categories          audio
+license             GPL-2
+maintainers         nomaintainer
+description         HE-AACv2/AAC+ encoder
+long_description    aacplusenc is based on 3GPP sources.
+
+homepage            http://teknoraver.net/software/mp4tools/
+# https://github.com/teknoraver/aacplusenc
+
+master_sites        http://ppa.launchpad.net/teknoraver/ppa/ubuntu/pool/main/a/${name}/
+extract.suffix      .tar.gz
+distname            ${name}_${version}
+
+checksums           rmd160  c59efc3a3f8126b4650b304a043f176b80078d82 \
+                    sha256  318df58a86ac0647d6bfbcd8766fe301ad2ed08ffb4ef4e016e088f4dceb1669 \
+                    size    547953
+
+depends_lib         path:lib/libfftw3f.dylib:fftw-3-single
+depends_test        bin:mplayer:MPlayer
+
+worksrcdir          aacplusenc
+
+patchfiles-append   configure.patch
+
+configure.env-append \
+                    PREFIX=${prefix}
+configure.pre_args  {}
+build.pre_args-append \
+                    EXTRACFLAGS=${configure.cflags}
+
+destroot {
+    # Don't use make install because it does not respect the common
+    # $(DESTDIR)$(prefix) and does not install the man page.
+    xinstall -m 0755 ${worksrcpath}/aacplusenc ${destroot}${prefix}/bin/aacplusenc
+    xinstall -m 0444 ${worksrcpath}/aacplusenc.1 ${destroot}${prefix}/share/man/man1/aacplusenc.1
+}
+
+test.run            yes
+test.target         test
+
+pre-test {
+    # install test.wav
+    # https://samplelib.com/sample-wav.html
+    file copy ${filespath}/sample-3s.wav ${worksrcpath}/test.wav
```

Comment:
I don't think we should package a .wav file into the ports tree.  Rather than keep it as a file in `./files`, it may make more sense to have it as a distfile that gets downloaded and _then_ copied into place for tests.

In order to prevent downloading for standard builds, we could probably move tests into a variant.

Reply:
How can a file only be downloaded for tests?

I don't understand what you mean by having a variant for tests.
Tests don't run for standard installs?

Reply:
Apologies. There were a lot of additional things that need to be clarified in that explanation.

1. We don't want to commit binary files like entire .wav files to the ports tree.  So the best solution here would be to download it at the same time that we're downloading the source code tarballs.  This means we specify the .wav file as an addition `distfile`.
2. If we aren't running tests, ie., just a standard install of `aacplusenc`, then we don't want to download the .wav file since we won't be using it.  However if the .wav file is specified as an additional `distfile`, then MacPorts will always download it everytime.
3. We get around the above by moving everything test-related into a _variant_. In this way, MacPorts will only download the .wav file and allow for tests to be run if and only if the variant is enabled.

Here's your Portfile modified to implement the above:

```
# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4

PortSystem          1.0

name                aacplusenc
version             0.17.5
categories          audio
license             GPL-2
maintainers         nomaintainer
description         HE-AACv2/AAC+ encoder
long_description    aacplusenc is based on 3GPP sources.

homepage            https://teknoraver.net/software/mp4tools/
# https://github.com/teknoraver/aacplusenc

master_sites        http://ppa.launchpad.net/teknoraver/ppa/ubuntu/pool/main/a/${name}/:launchpad
distname            ${name}_${version}

distfiles           ${name}_${version}${extract.suffix}:launchpad

checksums           ${name}_${version}${extract.suffix} \
                    rmd160  c59efc3a3f8126b4650b304a043f176b80078d82 \
                    sha256  318df58a86ac0647d6bfbcd8766fe301ad2ed08ffb4ef4e016e088f4dceb1669 \
                    size    547953

depends_lib         path:lib/libfftw3f.dylib:fftw-3-single
depends_test        bin:mplayer:MPlayer

worksrcdir          aacplusenc

patchfiles-append   configure.patch

configure.env-append \
                    PREFIX=${prefix}
configure.pre_args  {}
build.pre_args-append \
                    EXTRACFLAGS=${configure.cflags}

destroot {
    # Don't use make install because it does not respect the common
    # $(DESTDIR)$(prefix) and does not install the man page.
    xinstall -m 0755 ${worksrcpath}/aacplusenc ${destroot}${prefix}/bin/aacplusenc
    xinstall -m 0444 ${worksrcpath}/aacplusenc.1 ${destroot}${prefix}/share/man/man1/aacplusenc.1
}

variant test description {Enable testing} {
    test.run        yes
    test.target     test

    # Sample .wav file needed for tests. Download from Samplelib:
    master_sites-append https://download.samplelib.com/wav/:samplelib

    distfiles-append    sample-3s.wav:samplelib

    checksums-append    sample-3s.wav \
                        rmd160  37e56eacf829951d3a5c4586bafb6cc6613c3ce7 \
                        sha256  b3726eac5c9612ea20e245314812575bf9df5fb6b8024b80c7cfe9033452bb2b \
                        size    563756

    post-extract {
        # Install .wav file as test.wav
        file copy ${distpath}/sample-3s.wav ${worksrcpath}/test.wav
    }
}
```

Doing it this way means the variant needs to be enabled in order to run the tests:

```sh
sudo port test aacplusenc +test
```

Reply:
@herbygillot I applied your changes.

Is it possible to tell MacPorts not to extract the WAV file?

```
--->  Extracting sample-3s.wav
Executing:  cd "/opt/local/var/macports/build/_macports-ports_audio_aacplusenc/aacplusenc/work" && /usr/bin/gzip -dc '/opt/local/var/macports/distfiles/aacplusenc/sample-3s.wav' | /usr/bin/tar -xf -
gzip: /opt/local/var/macports/distfiles/aacplusenc/sample-3s.wav: not in gzip format
```

Reply:
> Is it possible to tell MacPorts not to extract the WAV file?

I think that might be doable with `extract.only`:

```
extract.only        ${distname}${extract.suffix}
```

Reply:
Seems to work, thanks.